hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -511,6 +516,17 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 			return ctrl.Result{}, fmt.Errorf("failed to reconcile AWSMachineTemplate: %w", err)
 		}
 		span.AddEvent("reconciled awsmachinetemplate", trace.WithAttributes(attribute.String("name", machineTemplate.GetName())))
+	case hyperv1.AgentPlatform:
+		machineTemplate = AgentMachineTemplate(nodePool, controlPlaneNamespace)
+		if err := r.Get(ctx, client.ObjectKeyFromObject(machineTemplate), machineTemplate); err != nil {
```

Comment:
Why not use CreateOrUpdate? The current name-changing approach means that we orphan old templates on config change

Reply:
Again, I copied this from AWS, see reconcileAWSMachineTemplate() in the same file. It does get and create.

Reply:
Templates can't just be change inline as that would break rolling upgrades. This needs to be solved in a provider agnostic way and it is tracked here https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/nodepool/nodepool_controller.go#L606

Let's tackle this aside from this PR.

Reply:
This needs to be consolidated with some of the logic it's in reconcileAWSMachineTemplate